### PR TITLE
[Telegram Action] Send base64 image with Telegram message

### PR DESCRIPTION
**Pull request by BasvanH**
I have extended the Telegram action with sending an base64 image from rules. This comes in handy when you have images stored in an Image Item.

### CODE REVIEW

File: bundles/action/org.openhab.action.telegram/README.md [hunk 1]
```diff
@@ -112,3 +112,17 @@ end
 Do not use username/password in url like in this example `http://<username>:<password>@server/image.png`; pass the credentials to the `sendTelegramPhoto` method instead.
 
 `http` and `https` are the only protocols allowed.
+
+In case you want to send a base64 jpeg or png image.
```
Comment (bgilmer77) [inline bundles/action/org.openhab.action.telegram/README.md]: I know this suggestion does not conform to the way the rest of the article is written, but please consider, "If you would like to send a base64..."

File: bundles/action/org.openhab.action.telegram/src/main/java/org/openhab/action/telegram/internal/Telegram.java [hunk 2]
```diff
@@ -181,50 +181,59 @@ static public boolean sendTelegramPhoto(@ParamDoc(name = "group") String group,
...
-            logger.warn("HTTP protocol violation: {}", e);
-            return false;
-        } catch (IOException e) {
-            logger.warn("Transport error: {}", e);
-            return false;
-        } finally {
-            getMethod.releaseConnection();
+        } else {
+            // Load image from provided base64 image
+            logger.warn("Photo base64 provided, converting to binary.");
```
Comment (bgilmer77) [inline bundles/action/org.openhab.action.telegram/src/main/java/org/openhab/action/telegram/internal/Telegram.java]: Please consider, "Photo base64 provided; converting to binary."  <- I changed the comma to a semicolon.

**Comment by BasvanH** [comment]
I'm fast, but not that fast :-)

File: bundles/action/org.openhab.action.telegram/src/main/java/org/openhab/action/telegram/internal/Telegram.java [hunk 3]
```diff
@@ -181,50 +181,59 @@ static public boolean sendTelegramPhoto(@ParamDoc(name = "group") String group,
...
-            logger.warn("HTTP protocol violation: {}", e);
-            return false;
-        } catch (IOException e) {
-            logger.warn("Transport error: {}", e);
-            return false;
-        } finally {
-            getMethod.releaseConnection();
+        } else {
+            // Load image from provided base64 image
+            logger.warn("Photo base64 provided; converting to binary.");
```
Comment (9037568) [inline bundles/action/org.openhab.action.telegram/src/main/java/org/openhab/action/telegram/internal/Telegram.java]: DEBUG, not WARN.

File: bundles/action/org.openhab.action.telegram/src/main/java/org/openhab/action/telegram/internal/Telegram.java [hunk 4]
```diff
@@ -181,50 +181,59 @@ static public boolean sendTelegramPhoto(@ParamDoc(name = "group") String group,
...
+            String base64Image = photoURL.split(",")[1];
```
Comment (9037568) [inline bundles/action/org.openhab.action.telegram/src/main/java/org/openhab/action/telegram/internal/Telegram.java]: Need to check your array bounds here.

File: bundles/action/org.openhab.action.telegram/src/main/java/org/openhab/action/telegram/internal/Telegram.java [hunk 5]
```diff
@@ -181,50 +181,59 @@ static public boolean sendTelegramPhoto(@ParamDoc(name = "group") String group,
...
+            image = javax.xml.bind.DatatypeConverter.parseBase64Binary(base64Image);
```
Comment (9037568) [inline bundles/action/org.openhab.action.telegram/src/main/java/org/openhab/action/telegram/internal/Telegram.java]: Need to wrap this in a try block.

File: bundles/action/org.openhab.action.telegram/src/main/java/org/openhab/action/telegram/internal/Telegram.java [hunk 6]
```diff
@@ -1,6 +1,6 @@
 /**
  * Copyright (c) 2010-2017 by the respective copyright holders.
```
Comment (9037568) [inline bundles/action/org.openhab.action.telegram/src/main/java/org/openhab/action/telegram/internal/Telegram.java]: Please change 2017 to 2018 here.

**Comment by BasvanH** [comment]
Could you please explain short how did you find the cause? This way I can do it myself in the future. Thanks.

**Comment by BasvanH** [comment]
Build failing with same error as previous build attempt:
`stderr: fatal: unable to access 'https://github.com/openhab/openhab1-addons/': Peer reports incompatible or unsupported protocol version.`

**Comment by 9037568** [comment]
This error looks, as you said before, as if it has nothing to do with these changes. Appears to be some sort of Jenkins plugin failure.

````
16:02:54.083  |  	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1996)
````

**Comment by 9037568** [comment]
Thanks, @BasvanH !

**Pull request merged**